tests/unit_test/test_curve.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import pytest
+import unittest
+from beprof.curve import Curve
+
+
+class TestCurveInit(unittest.TestCase):
+    """
+    Testing Curve initialization and .x .y
+    """
+    def test_list_init(self):
+        simple_lists = [[-12, 1], [-1, 7], [0, 3], [3, 17]]
+        c = Curve(simple_lists)
+        assert list(c.x) == [e[0] for e in simple_lists]
+        assert list(c.y) == [e[1] for e in simple_lists]
+
+    def test_numpy_array_init(self):
+        numpy_array = np.array([[-12, 1], [-1, 7], [0, 3], [3, 17]])
+        c = Curve(numpy_array)
+        assert c.x.tolist() == numpy_array[:, 0].tolist()
+        assert c.y.tolist() == numpy_array[:, 1].tolist()
+
+    def test_numpy_view_init(self):
+        # todo
+        pass
+
+    def test_empty_init(self):
+        with pytest.raises(TypeError):
+            Curve()
+        with pytest.raises(IndexError):
+            Curve([])
+
+    def test_one_point_init(self):
+        c = Curve([[1, 2]])
+        assert c.x == 1
+        assert c.y == 2
+
+    def test_two_point_init(self):
+        c = Curve([[-1, 3], [1, 7]])
+        assert c.x.tolist() == [-1, 1]
+        assert c.y.tolist() == [3, 7]
+
+
+class TestCurveRescale(unittest.TestCase):
+    def setUp(self):
+        # two the same Curves - one for modification, one for comparison
```

Comment:
Be more specific or use more meaningful names.
Which one is for modification and which for comparison ?